### DISCUSSION TITLE
osu!lazer has lost all of my beatmaps

### DISCUSSION THREAD

**Discussion by machRS** [ppy/osu#24090] [Q&A] [answered]
### Type

Game behaviour

### Bug description

Problem Summary: osu!lazer has lost all of my beatmaps. I suspect it could have a corrupted DB.

Problem Description:
Running latest version of osu!lazer.
About a fortnight ago, I was downloading new beatmaps through the client, when I experienced a crash to Desktop.
Any attempt to launch the game from then on failed. I rebooted my PC, and when the game launched, the client remembered my sign-in info and tablet settings, but lost all of my beatmaps & local scores. I was able to download new maps through the client, however I couldn't see old ones.
I've confirmed the data is still in the same directory using WizTree C:\Users\%username%\AppData\Roaming\osu

Troubleshooting steps attempted:
I tried uninstalling osu!lazer via appwiz.cpl, and then downloaded a fresh copy from the osu site, and reinstalled it. 
The client knew who I was on launch (must have grabbed the old config file), so I renamed the old folder from \roaming\osu\ to \roaming\osu.old\ and then launched the game, and tried using the "select an old install of osu" to reimport the files, however that feature wouldnt allow me to select that folder as a directory.
I've also tried restoring all recently deleted beatmaps, as well as all hidden difficulties to no luck.

To note:
I do not have osu! (legacy version) installed on my computer at all. Only osu!Lazer.
First time GH poster. Apologies if I have missed anything.

System Info:
In screenshots, but running Win11 Pro on latest updates, on Ryzen 5800 chip.

### Screenshots or videos

<img width="295" alt="image" src="https://github.com/ppy/osu/assets/138209109/502b295e-9813-44a2-8d73-e6b0cfd9f20b">
<img width="699" alt="image" src="https://github.com/ppy/osu/assets/138209109/15a52639-46bf-44ad-bec1-0c9822da28d2">
<img width="756" alt="image" src="https://github.com/ppy/osu/assets/138209109/8bd3817b-7af6-49cb-97d9-9972837e57e7">

### Version

2023.621.0

### Logs

[network.log](https://github.com/ppy/osu/files/11915293/network.log)
[performance.log](https://github.com/ppy/osu/files/11915295/performance.log)
[runtime.log](https://github.com/ppy/osu/files/11915296/runtime.log)
[updater.log](https://github.com/ppy/osu/files/11915297/updater.log)
[database.log](https://github.com/ppy/osu/files/11915299/database.log)
[input.log](https://github.com/ppy/osu/files/11915300/input.log)

**Comment by peppy** [comment]
Are you able to check your osu! folder for `client.realm` files? If you find one with a large filesize, can you first:

- Make a backup. This is all your data
- Delete `client.realm` and rename the largest version to `client.realm` replacing it
- Run the game and see if things recover

If this doesn't work, could you please zip the file and attach to this issue? I will look into it for you. If it's too big or you don't want to share your collections / beatmaps / score publicly, please email it to pe@ppy.sh

**Comment by machRS** [comment]
Screenshot of gamedir:

<img width="593" alt="image" src="https://github.com/ppy/osu/assets/138209109/3b27a7c9-7ea1-48bc-981c-316915521d96">

**Confirming I have completed below steps:**
Cut all files starting with "client*" from the root of the gamedir to a seperate backup location
Started the game
Game started into setup wizard, post completion no beatmaps were available.
Saw that there was two files client_(number)_corrupt.realm in the backup I created
Tried pasting the files starting with client_(number)_corrupt.realm back into the directory, and generating new client.realm* files to see if the game would recognise/process the corrupt realm files. No luck. 

I have sent you an email with the attachments, will have subject line: "osu!lazer has lost all of my beatmaps #24090". 

Appreciate you having a look, thankyou.

**Comment by machRS** [comment, marked as answer]
Hey Peppy, you can mark this as closed. Once you gave me a hint with the client.realm file, I searched the discussions and found one where you mentioned renaming the _corrupt.realm file and launching with it, and my game seems to have recovered back to where it was. If I find anything else i'll reopen the thread. If there is any further testing you would like me to do, im happy to help.

**Reply by integerrr** [reply]
that's what he meant when he asked you to rename the largest realm file to `client.realm`, which in your case would be `client_1687487639 _corrupt.realm`
